src/mptool.f90

```diff
@@ -0,0 +1,331 @@
+!Copyright (c) 2012-2022, Xcompact3d
+!This file is part of Xcompact3d (xcompact3d.com)
+!SPDX-License-Identifier: BSD 3-Clause
+module mptool
+  !
```

Comment:
We could add `use mpi` here and remove it from all the subroutines / functions in the module then.